public/components/register_model/__tests__/model_file_uploader_manager.test.ts

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { waitFor } from '@testing-library/dom';
+import { Model } from '../../../../public/apis/model';
+import { ModelFileUploadManager } from '../model_file_upload_manager';
+
+describe('ModelFileUploadManager', () => {
+  const uploadChunkMock = jest.fn();
+
+  beforeEach(() => {
+    jest.spyOn(Model.prototype, 'uploadChunk').mockImplementation(uploadChunkMock);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('should upload file by chunk', async () => {
+    const uploader = new ModelFileUploadManager();
+    const file = new File(['test model file'], 'model.zip', { type: 'application/zip' });
+    Object.defineProperty(file, 'size', { value: 30 * 1000 * 1000 });
+
+    uploader.upload({
+      modelId: 'test model id',
+      file,
+      chunkSize: 10 * 1000 * 1000,
+    });
+    await waitFor(() => expect(uploadChunkMock).toHaveBeenCalledTimes(3));
+  });
+
+  it('should call onUpdate', async () => {
+    const onUpdateMock = jest.fn();
+    const uploader = new ModelFileUploadManager();
+    const file = new File(['test model file'], 'model.zip', { type: 'application/zip' });
+    Object.defineProperty(file, 'size', { value: 30 * 1000 * 1000 });
+
+    uploader.upload({
+      modelId: 'test model id',
+      file,
+      chunkSize: 10 * 1000 * 1000,
+      onUpdate: onUpdateMock,
+    });
+    await waitFor(() => {
+      expect(onUpdateMock).toHaveBeenCalledWith({ total: 3, current: 1 });
+      expect(onUpdateMock).toHaveBeenCalledWith({ total: 3, current: 2 });
+      expect(onUpdateMock).toHaveBeenCalledWith({ total: 3, current: 3 });
+    });
+  });
+
+  it('should call onComplete', async () => {
+    const onCompleteMock = jest.fn();
+    const uploader = new ModelFileUploadManager();
+    const file = new File(['test model file'], 'model.zip', { type: 'application/zip' });
+    Object.defineProperty(file, 'size', { value: 30 * 1000 * 1000 });
+
+    uploader.upload({
+      modelId: 'test model id',
+      file,
+      chunkSize: 10 * 1000 * 1000,
+      onComplete: onCompleteMock,
+    });
+    await waitFor(() => expect(onCompleteMock).toHaveBeenCalled());
```

Comment:
It feels like testing `Observable` internals, because for `Observable`, `complete` is called in the end after all `next` calls finished. Did I get you correctly?

Reply:
Yes. I can get the point from the `rxjs`. Just thinking of do we need to write the unit test code according the implementation? In this case , we need to call onUpdate three times, then call onComplete. It's ok to just test onComplete here.

Reply:
I got your point, do you know how to assert one function is called after another?

Reply:
I don't have a particularly good solution, we can check the onUpload and onComplete execute order by controlling the Model.property.uploadChunk return promise resolved status.